pandas/tools/tests/test_merge_asof.py

```diff
@@ -452,6 +563,69 @@ def test_on_float(self):
 
         assert_frame_equal(result, expected)
 
+    def test_on_specialized_type(self):
+        # GH13936
+        for dtype in [np.uint16, np.uint32, np.uint64,
```

Comment:
are int8/uint8 not allowed (ok by me), just test that you raise on them (or do they work some other way)? same about float16 (should just raise)